sw/host/opentitanlib/src/io/jtag.rs

```diff
@@ -131,31 +131,270 @@ pub enum JtagTap {
     LcTap,
 }
 
-/// List of useful RISC-V general purpose registers
-#[derive(Clone, Copy, Debug, Deserialize, Serialize)]
-pub enum RiscvGpr {
-    GP,
-    SP,
-    RA,
+// Define an enum with an auto-generated `name` function that converts enum to string.
+//
+// For example, given
+// ```rust
+// def_reg_name! {
+//     #[derive(Clone, Copy, Debug)]
+//     pub enum Colour {
+//         Red,
+//         Green,
+//         Blue,
+//     }
+// }
+// ```
+//
+// This generates
+// ```rust
+// #[derive(Clone, Copy, Debug)]
+// pub enum Colour {
+//     Red,
+//     Gteen,
+//     Blue,
+// }
+//
+// impl Colour {
+//     fn name(self) -> &static str {
+//         match self {
+//             Self::Red => "red",
+//             Self::Green => "green",
+//             Self::Blue => "blue",
+//         }
+//     }
+// }
+// ```
+macro_rules! def_reg_name {
+    (
+        $(#[$attr:meta])*
+        $vis:vis enum $name:ident {
+            $($reg:ident,)*
+        }
+    ) => {paste::paste!{
+        // Emit the input as is.
+        $(#[$attr])*
+        $vis enum $name {
+            $($reg,)*
+        }
+
+        impl $name {
+            /// Get the name of the register in lowercase.
+            pub fn name(self) -> &'static str {
+                match self {
+                    $(Self::$reg => stringify!([<$reg:lower>]),)*
+                }
+            }
+        }
+    }}
 }
 
-/// List of useful RISC-V control and status registers
-#[derive(Clone, Copy, Debug, Deserialize, Serialize)]
-pub enum RiscvCsr {
-    DPC,
-    PMPCFG0,
-    PMPCFG1,
-    PMPCFG2,
-    PMPCFG3,
-    PMPADDR0,
-    PMPADDR15,
+def_reg_name! {
+    /// List of RISC-V general purpose registers
+    #[derive(Clone, Copy, Debug, Deserialize, Serialize)]
+    pub enum RiscvGpr {
```

Comment:
The proc-macros in the `strum` crate might be a more appropriate choice than creating our own macro:

```
use strum::IntoStaticStr;

#[derive(IntoStaticStr)]
#[strum(serialize_all = "lowercase")]
pub enum Foo {
  Foo,  Bar, Baz,
}
```

The strum crate has a number of different derive macros for transforming to string (IntoStaticStr, Display, ToString, etc).  You can pick whichever ones are appropriate to the use case.

Reply:
I didn't realize that strum is already one of our dependency. Will switch to that.

Reply:
Done.